components/TARGET_PSA/services/attestation/COMPONENT_PSA_SRV_IMPL/attest_crypto_keys.c

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (c) 2017-2019 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "tfm_plat_crypto_keys.h"
+#include "crypto.h"
+
+#include <stdlib.h>
+#include <math.h>
+
+
+#define ONE_BYTE (1u)
+
+/**
+ * \brief Copy the key to the destination buffer
+ *
+ * \param[out]  p_dst  Pointer to buffer where to store the key
+ * \param[in]   p_src  Pointer to the key
+ * \param[in]   size   Length of the key
+ */
+static inline void copy_key(uint8_t *p_dst, const uint8_t *p_src, size_t size)
+{
+    uint32_t i;
+
+    for (i = size; i > 0; i--) {
+        *p_dst = *p_src;
+        p_src++;
+        p_dst++;
+    }
+}
+
+static psa_status_t get_curve(psa_key_type_t type, enum ecc_curve_t *curve_type)
+{
+    psa_ecc_curve_t curve = PSA_KEY_TYPE_GET_CURVE(type);
+    switch (curve) {
+        case PSA_ECC_CURVE_SECP256R1:
+            *curve_type = P_256;
+            break;
+        case PSA_ECC_CURVE_SECP384R1:
+            *curve_type = P_384;
+            break;
+        case PSA_ECC_CURVE_SECP521R1:
+            *curve_type = P_521;
+            break;
+        case PSA_ECC_CURVE_CURVE25519:
+            *curve_type = X25519;
+            break;
+        case PSA_ECC_CURVE_CURVE448:
+            *curve_type = X448;
+            break;
+        default:
+            return (PSA_ERROR_NOT_SUPPORTED);
+    }
+
+    return PSA_SUCCESS;
+}
+
+enum tfm_plat_err_t
+tfm_plat_get_initial_attest_key(uint8_t          *key_buf,
+                                uint32_t          size,
+                                struct ecc_key_t *ecc_key,
+                                enum ecc_curve_t *curve_type)
+
+{
+    uint8_t *key_dst = NULL;
+    uint8_t *key_src;
+    uint32_t key_size;
+
+    psa_status_t crypto_ret;
+
+    uint8_t *public_key = NULL;
+    psa_key_type_t type;
+    psa_key_type_t public_type;
+    size_t bits;
+    size_t public_key_size = 0;
+    size_t public_key_length = 0;
+
+    uint32_t initial_attestation_public_x_key_size = 0;
+    uint32_t initial_attestation_public_y_key_size = 0;
+
+    const psa_key_id_t key_id = 17;
+    psa_key_handle_t handle = 0;
+
+    crypto_ret = psa_crypto_init();
+    if (crypto_ret != PSA_SUCCESS)
+    {
+        return TFM_PLAT_ERR_SYSTEM_ERR;
+    }
+
+    crypto_ret = psa_open_key(PSA_KEY_LIFETIME_PERSISTENT, key_id, &handle);
+    if (crypto_ret != PSA_SUCCESS)
+    {
+        return TFM_PLAT_ERR_SYSTEM_ERR;
```

Comment:
no crypto deinit?

Reply:
I removed all calling to `psa_crypto_init` - we assume init called before crypto operations, if not an error will be return.